File: unity/Assets/Scripts/Core/SlotEngine.cs
using UnityEngine;

namespace DragonGateSlot.Core
{
    public enum CellType { Card, Scatter }

    public struct Cell
    {
        public CellType Type;
        public int Value; // 1(A)~13(K), 0 for Scatter

        public override string ToString() =>
            Type == CellType.Scatter ? "SC" : Value.ToString();
    }

    public class SlotEngine : MonoBehaviour
    {
        const float ScatterRate = 0.05f;
        const int MinCard = 1;
        const int MaxCard = 13;

        Cell[,] _board = new Cell[3, 3]; // [row, col]

        public Cell[,] Board => _board;

        public Cell[,] Spin()
        {
            for (int r = 0; r < 3; r++)
                for (int c = 0; c < 3; c++)
                    _board[r, c] = GenerateCell();
            return _board;
        }

        Cell GenerateCell()
        {
            if (Random.value < ScatterRate)
                return new Cell { Type = CellType.Scatter, Value = 0 };
            return new Cell { Type = CellType.Card, Value = Random.Range(MinCard, MaxCard + 1) };
        }
    }
}

File: web/js/slotEngine.js
const SCATTER_RATE = 0.05;
const MIN_CARD = 1;
const MAX_CARD = 13;

export const CellType = { Card: 'Card', Scatter: 'Scatter' };

export function createCell(type, value) {
  return { type, value };
}

export function cellToString(cell) {
  if (cell.type === CellType.Scatter) return 'SC';
  const names = ['', 'A', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K'];
  return names[cell.value];
}

export function spin() {
  const board = [];
  for (let r = 0; r < 3; r++) {
    board[r] = [];
    for (let c = 0; c < 3; c++) {
      board[r][c] = generateCell();
    }
  }
  return board;
}

function generateCell() {
  if (Math.random() < SCATTER_RATE)
    return createCell(CellType.Scatter, 0);
  return createCell(CellType.Card, Math.floor(Math.random() * MAX_CARD) + MIN_CARD);
}
